### PR TITLE
doctest: changes several doctests that created Basics with non-Basic arguments

### PR DESCRIPTION
**Pull request by ThePauliPrinciple**
#### References to other Issues or PRs

Helps with https://github.com/sympy/sympy/pull/22445

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/series/sequences.py [hunk 1]
```diff
@@ -424,7 +424,8 @@ class SeqExpr(SeqBase):
 
     >>> from sympy.series.sequences import SeqExpr
     >>> from sympy.abc import x
-    >>> s = SeqExpr((1, 2, 3), (x, 0, 10))
+    >>> from sympy.core.containers import Tuple
+    >>> s = SeqExpr(Tuple(1, 2, 3), Tuple(x, 0, 10))
```
Comment (oscarbenjamin) [inline sympy/series/sequences.py:428]: Maybe this class should sympify its args

File: sympy/tensor/functions.py [hunk 2]
```diff
@@ -89,14 +89,14 @@ def shape(expr):
...
-    >>> from sympy import shape, Array, Matrix, Integral
+    >>> from sympy import shape, Array, ImmutableDenseMatrix, Integral
     >>> from sympy.abc import x
     >>> A = Array([1, 2])
     >>> shape(A)
     (2,)
     >>> shape(Integral(A, x))
     (2,)
-    >>> M = Matrix([1, 2])
+    >>> M = ImmutableDenseMatrix([1, 2])
```
Comment (oscarbenjamin) [inline sympy/tensor/functions.py:99]: Shouldn't this get sympified somewhere?

File: sympy/series/sequences.py (hunk 1 above)
Comment (ThePauliPrinciple) [reply sympy/series/sequences.py:428]: It actually doesn't have a `__new__` method at all and the docstring says `Various sequences should inherit from this class.` So I interpreted it as a class that should normally not be used by users (similar to `Basic`).

Comment (oscarbenjamin) [reply sympy/series/sequences.py:428]: Do the subclasses sympify their arguments?

File: sympy/tensor/functions.py (hunk 2 above)
Comment (ThePauliPrinciple) [reply sympy/tensor/functions.py:99]: It does in `Integral`, but not in `NewExpr` (user defined subclass). Here I thought the doc would be easier to read if `NewExpr` didn't have a `__new__` method defined

Comment (oscarbenjamin) [reply sympy/tensor/functions.py:99]: Ok, fair enough

File: sympy/series/sequences.py (hunk 1 above)
Comment (ThePauliPrinciple) [reply sympy/series/sequences.py:428]: Yes, there were also no tests that had to be changed for the subclasses as far as I can remember. They do use `sympify` over `_sympify` though (and are fairly complicated).

**Comment by oscarbenjamin** [comment]
Looks good

**Pull request merged**